File: android2_lesson03/src/java/java/com/example/ama/android2_lesson03/ui/search/base/SearchPresenter.java
package com.example.ama.android2_lesson03.ui.search.base;

import com.example.ama.android2_lesson03.ui.base.Presenter;
import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.Marker;

public interface SearchPresenter<T extends SearchOnTheMapView> extends Presenter<T> {
    void loadSavedState();

    void saveState(Marker currentMarker);

    void findAddressByQuery(String query);

    void findAddressByLatLng(LatLng latLng);

    void sendQueryToGMapsApp(Marker currentMarker, LatLng cameraPosition, float zoom);

    void findMyLocation();

    void onMarkerClick(Marker marker);

    void saveMarker(Marker currentMarker, String customName);

}

File: android2_lesson03/src/kotlin/java/com/example/ama/android2_lesson03/ui/search/base/SearchPresenter.kt
package com.example.ama.android2_lesson03.ui.search.base

import com.example.ama.android2_lesson03.ui.base.Presenter
import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.Marker

interface SearchPresenter<T : SearchOnTheMapView> : Presenter<T> {
    fun findAddressByQuery(query: String)
    fun findAddressByLatLng(latLng: LatLng)
    fun sendQueryToGMapsApp(currentMarker: Marker?, cameraPosition: LatLng?, zoom: Float?)
    fun findMyLocation()
    fun saveMarker(marker: Marker, customName: String)
    fun onMarkerClick(marker: Marker)
    fun saveState(currentMarker: Marker?)
    fun loadSavedState()
    fun subscribeOnLocationUpdates()
    fun unsubscribeOfLocationUpdates()
}